src/infrastructure/delivery_method.rs

```diff
@@ -0,0 +1,38 @@
+/// Sending method type
+#[derive(Copy, Clone, Debug,PartialOrd, PartialEq, Eq)]
+pub enum DeliveryMethod
+{
+    /// Unreliable. Packets can be dropped, duplicated or arrive without order
+    Unreliable,
+    /// Reliable. All packets will be sent and received, but without order
+    ReliableUnordered,
+    /// Unreliable. Packets can be dropped, but never duplicated and arrive in order
+    SequencedOrdered,
+    /// Unreliable. Packets can be dropped, and arrive out of order but you will be able to retrieve dropped packet.
+    SequencedUnordered,
+    /// Reliable and ordered. All packets will be sent and received in order
+    ReliableOrdered,
+}
+
+impl DeliveryMethod {
+    pub fn get_delivery_method_id(delivery_method: DeliveryMethod) -> u8 {
+        match delivery_method {
```

Comment:
you should be able to do `delivery_method as u8`
https://doc.rust-lang.org/beta/reference/items/enumerations.html#custom-discriminant-values-for-field-less-enumerations

Reply:
Thanks, of course, knew that.